components/admin_console/admin_definition.jsx

```diff
@@ -722,6 +722,106 @@ export default {
+                            label: t('admin.gitlab.clientIdTitle'),
+                            label_default: 'Application ID:',
+                            help_text: t('admin.gitlab.clientIdDescription'),
+                            help_text_default: 'Obtain this value via the instructions above for logging into GitLab',
+                            placeholder: t('admin.gitlab.clientIdExample'),
+                            placeholder_default: 'E.g.: "jcuS8PuvcpGhpgHhlcpT1Mx42pnqMxQY"',
+                            isDisabled: needsUtils.stateValueFalse('GitLabSettings.Enable'),
+                        },
+                        {
+                            type: Constants.SettingsTypes.TYPE_TEXT,
+                            key: 'GitLabSettings.Secret',
+                            label: t('admin.gitlab.clientSecretTitle'),
+                            label_default: 'Application Secret Key:',
+                            help_text: t('admin.gitlab.clientSecretDescription'),
+                            help_text_default: 'Obtain this value via the instructions above for logging into GitLab.',
+                            placeholder: t('admin.gitlab.clientSecretExample'),
+                            placeholder_default: 'E.g.: "jcuS8PuvcpGhpgHhlcpT1Mx42pnqMxQY"',
+                            isDisabled: needsUtils.stateValueFalse('GitLabSettings.Enable'),
+                        },
+                        {
+                            type: Constants.SettingsTypes.TYPE_TEXT,
+                            key: 'GitLabSettings.Url',
+                            label: t('admin.gitlab.siteUrl'),
+                            label_default: 'GitLab Site URL:',
+                            help_text: t('admin.gitlab.siteUrlDescription'),
+                            help_text_default: 'Enter the URL of your GitLab instance, e.g. https://example.com:3000. If your GitLab instance is not set up with SSL, start the URL with http:// instead of https://.',
+                            placeholder: t('admin.gitlab.siteUrlExample'),
+                            placeholder_default: 'E.g.: https://',
+                            isDisabled: needsUtils.stateValueFalse('GitLabSettings.Enable'),
+                        },
+                        {
+                            type: Constants.SettingsTypes.TYPE_TEXT,
+                            key: 'GitLabSettings.UserApiEndpoint',
+                            label: t('admin.gitlab.userTitle'),
+                            label_default: 'User API Endpoint:',
+                            dynamic_value: (value, config, state) => {
+                                if (state['GitLabSettings.Url']) {
+                                    return state['GitLabSettings.Url'] + '/api/v4/user';
+                                }
+                                return '';
+                            },
+                            isDisabled: () => true,
```

Comment:
nit: it would be nice if `isDisabled` could be set to `true`, and the framework would automatically determine if it's callable.

Reply:
This might work
```javascript
let isDisabled = schema.isDisabled;
if (typeof isDisabled === 'function') {
    isDisabled = isDisabled(args);
}
```